app/src/main/java/com/example/parkingsystemkotlin/activities/ParkingSpaceReservationActivity.kt

```diff
@@ -0,0 +1,46 @@
+package com.example.parkingsystemkotlin.activities
+
+import android.app.DatePickerDialog
+import android.app.TimePickerDialog
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.widget.DatePicker
+import android.widget.TimePicker
+import androidx.appcompat.app.AppCompatActivity
+import com.example.parkingsystemkotlin.databinding.ActivityReservationParkingSpaceBinding
+import com.example.parkingsystemkotlin.mvp.contracts.ParkingSpaceReservationContract
+import com.example.parkingsystemkotlin.mvp.model.ParkingSpaceReservationModel
+import com.example.parkingsystemkotlin.mvp.presenter.ParkingSpaceReservationPresenter
+import com.example.parkingsystemkotlin.mvp.view.ParkingSpaceReservationView
+
+class ParkingSpaceReservationActivity : AppCompatActivity(), DatePickerDialog.OnDateSetListener, TimePickerDialog.OnTimeSetListener {
+
+    private lateinit var binding: ActivityReservationParkingSpaceBinding
+    private lateinit var presenter: ParkingSpaceReservationContract.ParkingSpaceReservationPresenter
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityReservationParkingSpaceBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+        presenter = ParkingSpaceReservationPresenter(ParkingSpaceReservationModel(), ParkingSpaceReservationView(this, binding))
+        setListeners()
+    }
+
+    companion object {
```

Comment:
put the companion at the end of the class please

Reply:
Done!